src/main/kotlin/com/github/pemistahl/lingua/api/LanguageDetector.kt

```diff
@@ -229,41 +183,36 @@ class LanguageDetector internal constructor(
             }
         }
 
-        val languagesWithMinimumRequiredCharCountExist = languageCharCounts
+        return languageCharCounts
             .asSequence()
             .filter { it.value >= minimalRequiredCharCount }
-            .count() > 0
-
-        return if (languagesWithMinimumRequiredCharCountExist)
-            languageCharCounts.toList().sortedByDescending { it.second }.first().first
-        else
-            UNKNOWN
+            .maxBy { it.value }?.key
+            ?: UNKNOWN
```

Comment:
This improvement will also be more efficient as it avoids creating the list and sorting it